packages/fxa-settings/src/pages/Signin/SigninUnblock/container.tsx

```
@@ -0,0 +1,112 @@
+import { BeginSigninResponse, SigninContainerIntegration } from '../interfaces';
+
+import SigninUnblock from '.';
+import {
+  BeginSigninWithUnblockCodeHandler,
+  ResendUnblockCodeHandler,
+  SigninUnblockLocationState,
+} from './interfaces';
+import { handleGQLError } from '../utils';
+import { hardNavigateToContentServer } from 'fxa-react/lib/utils';
+
+const SigninUnblockContainer = ({
+  integration,
+}: { integration: SigninContainerIntegration } & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: SigninUnblockLocationState;
+  };
+  const { email, hasLinkedAccount, hasPassword, password } =
+    location.state || {};
+
+  const wantsTwoStepAuthentication =
+    isOAuthIntegration(integration) && integration.wantsTwoStepAuthentication();
+
+  const [beginSignin] = useMutation<BeginSigninResponse>(BEGIN_SIGNIN_MUTATION);
+
+  const signinWithUnblockCode: BeginSigninWithUnblockCodeHandler = async (
+    unblockCode: string
+  ) => {
+    // TODO in FXA-6518 oauth ticket
+    // const service = integration.getService();
+    const options = {
+      verificationMethod: VerificationMethods.EMAIL_OTP,
+      unblockCode,
+    };
+    // TODO in FXA-6518 oauth ticket
+    //   // keys must be true to receive keyFetchToken for oAuth and syncDesktop
+    //   keys: isOAuth || isSyncDesktopV3,
+    //   service: service !== MozServices.Default ? service : undefined,
+    // };
+    try {
+      const { authPW } = await getCredentials(email, password);
+      const { data } = await beginSignin({
+        variables: {
+          input: {
+            email,
+            authPW,
+            options,
+          },
+        },
+      });
+      return { data };
+    } catch (error) {
+      // TODO consider additional error handling - any non-gql errors will return an unexpected error
```

I think `beginSignin` will always be a GQL error, so the only other one here would be maybe from `getCredentials`, which we probably want to return an unexpected error? Probably fine to leave the comment though.